### PR TITLE
20540: Updates unit tests, casting in char_p_to_bytes

### PR DESCRIPTION
**Pull request by apbassett**

### CODE REVIEW

File: amalgam/api.py [hunk 1]
```diff
@@ -602,7 +602,7 @@ def str_to_char_p(
         buf.value = value
         return buf
 
-    def char_p_to_bytes(self, p: c_char_p) -> bytes | None:
+    def char_p_to_bytes(self, p: _Pointer[c_char_p] | c_char_p) -> bytes | None:
```
Comment (fulpm) [inline amalgam/api.py:605]:
```suggestion
    def char_p_to_bytes(self, p: _Pointer[c_char] | c_char_p) -> bytes | None:
```

**Review by fulpm — DISMISSED** [review]

File: amalgam/test/test_standalone.py [hunk 2]
```diff
@@ -9,21 +9,22 @@
 
 import amalgam
 from amalgam.api import Amalgam
-from amalgam.test import amalgam_path
 
 _logger = logging.getLogger(__name__)
 
-
-is_amalgam_installed = os.path.exists(amalgam_path)
+amlg_postfix = '-mt' if '-mt' in os.getenv('HOWSO_CONFIG', '') else '-st'
```
Comment (mkoistinen) [inline amalgam/test/test_standalone.py:15]: Are you requiring `-mt` to be in the name of the config file? Is that something we guarantee in the pipelines? Is this not expected to work on a dev machine?

Comment (fulpm) [reply amalgam/test/test_standalone.py:15]: Maybe rename this to something like `AMALGAM_LIBRARY_POSTFIX`

Comment (fulpm) [reply amalgam/test/test_standalone.py:15]: `HOWSO_CONFIG` is already used by howso-engine-py for defining the path to your howso.yaml file.

Comment (mkoistinen) [reply amalgam/test/test_standalone.py:15]: Not to mention, if you don't have `HOWSO_CONFIG` defined, Howso-Engine-Py will look here: `~/.howso/howso.yaml` and in the current directory for a `howso.yaml` file.

**Review by mkoistinen — APPROVED** [review]

**Pull request merged**